File: 118-pascals-triangle/pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function (numRows) {
    if (numRows === 0) return [[]];
    let tri = [[1]];

    for (let r = 1; r < numRows; r++) {
        let temp = [0, ...tri[r - 1], 0];
        let row = [];
        for (let i = 1; i < temp.length; i++) {
            row.push(temp[i - 1] + temp[i]);
        }
        tri.push(row);
    }
    return tri;
};

//nRows = 3
//tri = [[1], [1, 1], [1, 2, 1]]
//temp = [0, 1, 1, 0]
//row = [1, 2, 1]

File: 118-pascals-triangle/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 0:
            return [[]]
        
        tri = [[1]]

        for r in range(1, numRows):
            print("r=", r)
            temp = [0] + tri[r - 1] + [0]
            row = []
            
            for i in range(1, len(temp)):
                row.append(temp[i - 1] + temp[i])
            
            tri.append(row)
        
        return tri


